playbooks/group_vars/all/octavia.yml

```diff
@@ -33,7 +33,7 @@ octavia_rabbitmq_host_group: "{{ rabbitmq_host_group }}"
 # Overwrite haproxy
 haproxy_octavia_whitelist_networks: "{{ haproxy_whitelist_networks }}"
 
-haproxy_extra_services:
+haproxy_service_configs:
```

Comment:
This is overwriting *all* of the haproxy configurations, which is not correct. See https://github.com/openstack/openstack-ansible/blob/stable/newton/playbooks/haproxy-install.yml#L63

Using haproxy_extra_services is the correct method.

Can you please provide more detail about why this is being done?

Reply:
It might be that RPC1 is uniquely broken but we had to resort to this to get the haproxy stanzas deployed. My theory is that the definition of haproxy_extra_services  in /etc/openstack_deploy/user_rpco_variables_overrides.yml which gets sourced with -e if you run openstack_ansible overrides whatever we define in our -e file... This route has been taken successfully by our designate and k8-tools deploys as well.

Reply:
Here is the definition:
# Haproxy config for TIGK grafana
haproxy_extra_services:
  - service:
      haproxy_service_name: tigk-grafana
      haproxy_backend_nodes: "{{ groups['influx_hosts'] }}"
      haproxy_ssl: "{{ haproxy_ssl }}"
      haproxy_port: 8089
      haproxy_balance_type: http

Reply:
Hmm, so if you do what you're doing here it works - but you risk your config being removed when the next RPC-O deployment is done. It's probably a good idea to validate whether that happens or not before making further changes.

Reply:
@odyssey4me : Hmm, so if you do what you're doing here it works - but you risk your config being removed when the next RPC-O deployment is done. It's probably a good idea to validate whether that happens or not before making further changes.